src/models/methods/ledger_data.ts

```diff
@@ -0,0 +1,27 @@
+import { LedgerIndex } from "../common";
+import { LedgerEntry } from "../ledger";
+import { BaseRequest, BaseResponse } from "./baseMethod";
+
+export interface LedgerDataRequest extends BaseRequest {
+    command: "ledger_data"
+    ledger_hash?: string
+    ledger_index?: LedgerIndex
+    binary?: boolean
+    limit?: number
+    marker?: any
+}
+
+interface LedgerDataAdditionalFields {
+    data?: string
+    ledgerEntryType?: string
+    index: string
+}
+
+export interface LedgerDataResponse extends BaseResponse {
+    result: {
+        ledger_index: number
+        ledger_hash: string
+        state: (LedgerDataAdditionalFields & LedgerEntry)[]
```

Comment:
This would be better typed as (paraphrased) `{index: string} |({data: string} | ({ledgerEntryType: string} & LedgerEntry)
See the examples for a better understanding
Honestly it would be better to create a (non-exported) separate type that goes here